test/HelperPluginManagerTest.php

```diff
@@ -55,10 +56,10 @@ public function testNoRendererInjectedInHelperWhenRendererIsNotPresent()
 
     public function testRegisteringInvalidHelperRaisesException()
     {
-        $helpers = new HelperPluginManager(new ServiceManager(), ['services' => [
-            'test' => $this,
+        $helpers = new HelperPluginManager(new ServiceManager(), ['factories' => [
+            'test' => function () { return $this; },
         ]]);
-        $this->setExpectedException('Zend\ServiceManager\Exception\InvalidServiceException');
+        $this->setExpectedException('Zend\View\Exception\InvalidHelperException');
```

Comment:
This will need to check for a different exception depending on whether it's run under v2 or v3. Easiest way is to call a `getInvalidServiceException()` method in the same way the `CommonPluginManagerTrait` does. Or, if it's not a whole load of retooling, use the trait in this test and ditch the compatibility test.